slides/cen-first-year-wg-sep2023.qmd

```diff
@@ -0,0 +1,209 @@
+- Start of video series "Statistical Software Engineering 101" [🌎](https://www.youtube.com/playlist?list=PL848NFA2PWgCR35n02yn1ZV7JqSu3NMxS)
+  - currently 2 videos, hopefully we can still produce more content
+
+## Conference contributions and Publications
+
+- Dedicated sessions with discussions at ISCB, CEN (now), ASA/FDA workshop
+- Presentations at PSI, JSM, Pharma RUG, BBS, etc. [🌎](https://rconsortium.github.io/asa-biop-swe-wg/presentations.html)
+- BIOP Report [🌎](https://rconsortium.github.io/asa-biop-swe-wg/blog/biop_report/)
+- Blog [🌎](https://rconsortium.github.io/asa-biop-swe-wg/blog.html)
+
+# Ingredients for successful and sustainable collaboration
+
+## Human factors
+
+- Mutual interest
+- Mutual trust
+- Prerequisite is getting to know each other
+  - Although mostly just online, biweekly calls help a lot with this
+- Reciprocity mindset
+  - First give away something, and more will come back to you
+
+## Development process
+
+- Different perspectives in discussions and code review help to optimize the user interface and thus experience
+- Important to go public as soon as possible 
+  - don't wait for the product to be finished
+  - you never know who else might be interested/could help
+- Version control with git
+  - cornerstone of effective collaboration
+
+## Coding standards
+
+- Consistent and readable code style simplifies joint work
+- Written (!) contribution guidelines
+- Lowering the entry hurdle using developer calls
+
+## Robust test suite
+
+- Unit tests are essential for preventing regression and assuring quality
+- Especially with compiled code critical to see if package works correctly
+- Use continuous integration during development to make sure nothing breaks along the way
+
+## Documentation
+
+- Lots of work but extremely important 
+  - often need to start with methods details write up before being able to code
+- Needs to be kept up-to-date
+- Need to have examples & vignettes
+  - Testing alone is not sufficient 
+  - Builds trust with users
+  - Reference for developers over time
+
+# Long term vision
+
+## Statisticians have software engineering skills
+
+- These skills are taught at university
+- Statisticians can use basic practices in their daily work ...
+- ... to ensure reproducibility of statistical analyses and research results
+  
+## Innovation does not stop with publication
+
+- Methods research does not end when the first methods paper is published
+- Initial prototype code as paper supplement is not sufficient
+- Continue to developing open source and reliably tested software packages ...
+- ... to enable users to easily use the new methodology in their own applications
+
+## Industry develops common code base
+
+- Increasingly companies work in the open source as much as possible
+- Rather than repeating similar developments internally ...
+- ... to become more cost-effective and transparent towards society
+
+# Next steps
+
+## Next steps
```

Comment:
Since we have quite many bullet points in next steps, maybe we could consider have a few categories first e.g. R package development, dissemination/discussion, outreach/collaboration etc, then put the bullet points into those categories.

Reply:
thanks @ywang-gilead, good idea :-)